### PR TITLE
Optional low resolution source maps

### PR DESCRIPTION
**Pull request by crisptrutski**
In many situations most of the value in a source map is given by the set of all first-in-their-line mappings.

In large builds, the cost of generating source maps is considerable, and emitting only this "blurrier" source map seems to be a good trade-off.

@johnjbarton please confirm if https://github.com/crisptrutski/traceur-compiler/commit/68b58bdcc86dbc02dbd6b660cb85684e428aa60a was a bug or not

### CODE REVIEW

**Comment by johnjbarton** [comment]
> @johnjbarton please confirm if crisptrutski@68b58bd was a bug or not
Yes, a bug. I've been trying to kill off `toSource()`, I think it is only used in some tests now and those probably don't use sourcemaps.  Did you find this by a failure or by inspection?

File: src/Compiler.js [hunk 1]
```diff
@@ -263,6 +263,8 @@ export class Compiler {
     this.sourceMapConfiguration_ =
         this.createSourceMapConfiguration_(outputName, sourceRoot);
     if (this.sourceMapConfiguration_) {
+      this.sourceMapConfiguration_.lowResolution =
+        this.options_.lowResolutionSourceMap;
```
Comment (johnjbarton) [inline src/Compiler.js]: 4 spaces when breaking statements (older code may still use 2 some places)

**Comment by crisptrutski** [comment]
Just through inspection, no pesky use case here

File: src/Options.js [hunk 2]
```diff
@@ -164,7 +165,7 @@ export class Options {
   }
 
   get sourceMaps() {
-    return this.sourceMaps_;
+    return this.lowResolutionSourceMap || this.sourceMaps_;
```
Comment (johnjbarton) [inline src/Options.js]: If `this.lowResolutionSourceMap` is true, then we will return `true` but that is not valid value for this property correct?

Comment (crisptrutski) [inline src/Options.js]: Thought that maybe this option should implictly enable source maps. Soured on the idea though, lets scrap it.

**Comment by johnjbarton** [comment]
I think we should have a test for this option, verifying that it produces a valid map similar to our existing sourcemap tests.

File: src/Options.js (hunk 2 above)
Comment (johnjbarton) [inline src/Options.js]: As you probably know, the sourcemaps option controls the output format, inline, memory, or file. I suppose we could have the same values on lowResolutionSourceMap but I'm not keen on more tricky options code.

Comment (crisptrutski) [inline src/Options.js]: Lets keep the options orthogonal, and ignore `lowResolutionSourceMap` if `sourceMaps` is not set.

Forgot that the `sourceMaps` option was not just a boolean, so definitely backing this line out.

**Comment by crisptrutski** [comment]
@johnjbarton backed out the noise, let me know if you want the two tests DRYed up

**Comment by johnjbarton** [comment]
Sorry: please add an entry to the src/Options.js descriptions and check how it reads when you run `./tracuer`.  Then LGTM

**Comment by crisptrutski** [comment]
@johnjbarton thanks for vetting this so quickly - CLI doc string added

**Comment by arv** [comment]
LGTM

Squashing and committing

**Comment by arv** [comment]
Actually, no need to squash.

**Comment by arv** [comment]
Rebased and commited

**Pull request closed without merging**